MapleServer2/Triggers/NpcTriggers.cs

```diff
@@ -1,12 +1,30 @@
-﻿using System.Numerics;
+﻿using System.Linq;
+using System.Numerics;
 using Maple2.Trigger.Enum;
+using Maple2Storage.Types.Metadata;
+using MapleServer2.Data.Static;
+using MapleServer2.Packets;
+using MapleServer2.Types;
 
 namespace MapleServer2.Triggers
 {
     public partial class TriggerContext
     {
-        public void AddBalloonTalk(int spawnPointId, string msg, int duration, int delayTick, bool npcId)
-        {
+        public void AddBalloonTalk(int spawnPointId, string msg, int duration, int delayTick, bool isNpcId)
+        {
+            if (!isNpcId)
+            {
+                if (spawnPointId == 0)
+                {

```

Comment:
You could merge this two ifs

Reply:
or double invert and return early

Reply:
This isn't complete so I did not want to merge or do two inverts/return.
There will be cases where it's a mix of the two
isNpcId = true &  spawnPointId = 0
isNpcId = true & spawnPointId != 0
isNpcId = false & spawnPointId = 0
isNpcId = false & spawnPointId != 0